app/services/hashtag_autocomplete_service.rb

```diff
@@ -130,6 +135,20 @@ def initialize(guardian)
     @guardian = guardian
   end
 
+  def find_by_ids(ids_by_type)
+    HashtagAutocompleteService
+      .data_source_types
+      .each_with_object({}) do |type, hash|
+        data_source = HashtagAutocompleteService.data_source_from_type(type)
+        next if !data_source || !data_source.respond_to?(:find_by_ids)
+
+        hashtags = data_source.find_by_ids(guardian, ids_by_type[type])
```

Comment:
There is a bug here; since you are looping over all types regardless of whether it is included in `ids_by_type`, this call `ids_by_type[type]` can be sometimes nil.

Reply:
We should add a general test for this `find_by_ids` method in `HashtagAutocompleteService` too

Reply:
Nice catch, although it did not affect any of the current `find_by_ids` methods because they all seem to work when `nil` is passed. 😅

Reply:
The chat one didn't for me, it errorred.